File: ddpoliglot_v6_flutter/lib/models/data/MixItem.dart
// ignore_for_file: file_names
import 'package:json_annotation/json_annotation.dart';

part 'MixItem.g.dart';

@JsonSerializable()
class MixItem {
  int mixItemID;
  int mixParamID;
  String? keyString;
  String? source;
  String? inDict;
  String? inContext;
  String? childrenType;
  bool endPhrase;
  bool pretext;
  int orderNum;
  bool baseWord;

  MixItem(
    this.mixItemID,
    this.mixParamID,
    this.keyString,
    this.source,
    this.inDict,
    this.inContext,
    this.childrenType,
    this.endPhrase,
    this.pretext,
    this.orderNum,
    this.baseWord,
  );

  factory MixItem.fromJson(Map<String, dynamic> data) =>
      _$MixItemFromJson(data);

  Map<String, dynamic> toJson() => _$MixItemToJson(this);
}

File: ddPoliglotV6.Data/Models/MixItem.cs

using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace ddPoliglotV6.Data.Models
{
    public class MixItem
    {
        public int MixItemID { get; set; }

        public int MixParamID { get; set; }

        [MaxLength(50)]
        public string KeyGuid { get; set; }

        [MaxLength(200)]
        public string Source { get; set; }

        [MaxLength(300)]
        public string InDict { get; set; }

        [MaxLength(300)]
        public string InContext { get; set; }

        [MaxLength(50)]
        public string ChildrenType { get; set; }

        public bool EndPhrase { get; set; }
        public bool Pretext { get; set; }
        public int OrderNum { get; set; }
        public bool baseWord { get; set; }
    }
}
